crates/angr/src/segmentlist.rs

```diff
@@ -183,6 +183,10 @@ impl SegmentList {
         if size == 0 {
             return;
         }
+        // ensure address + size does not overflow
+        if address.checked_add(size).is_none() {
+            return;
```

Comment:
Returning silently seems like a footgun- I think this should either be an error or even better, change the api to `[start, end)` instead of address+size, which both mirrors the rangemap api and makes this error impossible.

Reply:
I’m fine with either solution. Feel free to implement what you propose.